web/server/entrypoint.go

```diff
@@ -417,8 +433,15 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 		err = multierr.Combine(err, watcher.Close())
 	}()
 	onWatchDone := make(chan struct{})
-	oldCfg := processedConfig
+	// Use `fullProcessedConfig` as the initial `oldCfg` for the config watcher
+	// goroutine, as we want incoming config changes to be compared to the full
+	// config.
+	oldCfg := fullProcessedConfig
 	utils.ManagedGo(func() {
```

Comment:
It's probably about time this lambda gets its own function/name. I think a lot of my above concern goes away if this 60 lines of control flow keyword soup is hidden by default.

Reply:
Great idea; working on it + will re-request review when done.

Reply:
Done; maybe.